src/app/news/news-list.component.ts

```diff
@@ -197,4 +195,35 @@ export class NewsListComponent implements OnInit, OnChanges {
     return item._id;
   }
 
+  getCurrentItems(): any[] {
+    if (this.replyViewing._id === 'root') {
+      return this.items.filter(item => !item.doc.replyTo);
+    }
+    return this.replyObject[this.replyViewing._id] || [];
+  }
+
+  paginateItems(list: any[], start: number, size: number) {
+    const end = start + size;
+    const page = list.slice(start, end);
+    return {
+      items: page,
+      endIndex: start + page.length,
+      hasMore: end < list.length
+    };
+  }
+
+  loadPagedItems(initial = true) {
+    const news = this.getCurrentItems();
```

Comment:
Would be a little better to set this to a variable (`this.filteredNews`, maybe?) rather than running this filter every time "Show More" is clicked, I think.